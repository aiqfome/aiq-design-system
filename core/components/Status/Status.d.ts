import React, { ReactNode } from 'react';
export interface Props {
    children?: ReactNode;
    statusColor?: string;
}
export declare const Status: React.FC<Props>;
//# sourceMappingURL=Status.d.ts.map