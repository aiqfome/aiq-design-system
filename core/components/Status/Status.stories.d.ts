import React from 'react';
declare const _default: {
    component: React.FC<import("./Status").Props>;
    title: string;
};
export default _default;
export declare const Basic: React.FC;
//# sourceMappingURL=Status.stories.d.ts.map