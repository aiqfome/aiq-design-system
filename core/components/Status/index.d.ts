export * from './Status';
//# sourceMappingURL=index.d.ts.map