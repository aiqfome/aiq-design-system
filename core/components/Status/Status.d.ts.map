{"version":3,"file":"Status.d.ts","sourceRoot":"","sources":["../../../src/components/Status/Status.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,EAAE,EAAE,SAAS,EAAE,MAAM,OAAO,CAAA;AAOxC,MAAM,WAAW,KAAK;IACpB,QAAQ,CAAC,EAAE,SAAS,CAAA;IACpB,WAAW,CAAC,EAAE,MAAM,CAAA;CACrB;AAeD,eAAO,MAAM,MAAM,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAElC,CAAA"}