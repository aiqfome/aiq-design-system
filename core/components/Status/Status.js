"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Status = void 0;
const react_1 = __importDefault(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importDefault(require("styled-components"));
const Text_1 = require("../Text");
const StatusStyled = styled_components_1.default(Text_1.Text) `
  &::before {
    content: '';
    display: inline-block;
    width: 8px;
    height: 8px;
    margin-right: 10px;
    border-radius: 50%;
    background-color: ${({ statusColor, theme }) => statusColor ? theme.colors[statusColor] : '#dedede'};
  }
`;
exports.Status = (_a) => {
    var { children } = _a, props = __rest(_a, ["children"]);
    return react_1.default.createElement(StatusStyled, Object.assign({}, props), children);
};
exports.Status.propTypes = {
    children: prop_types_1.default.node,
    statusColor: prop_types_1.default.string
};
