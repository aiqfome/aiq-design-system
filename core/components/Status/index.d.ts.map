{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/Status/index.tsx"],"names":[],"mappings":"AAAA,cAAc,UAAU,CAAA"}