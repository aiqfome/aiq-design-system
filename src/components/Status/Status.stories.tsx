import React, { ReactElement } from 'react'

import { Flex } from '../Flex'
import { Status } from './Status'

export default {
  component: Status,
  title: 'Status'
}

export const Basic: React.FC = (): ReactElement => (
  <Flex flexDirection='column' variant='fullCentralized'>
    <Status>status default</Status>
    <Status statusColor='primary'>primary</Status>
    <Status statusColor='success'>success</Status>
    <Status statusColor='info'>information</Status>
    <Status statusColor='warning'>warning</Status>
    <Status statusColor='error'>error</Status>
  </Flex>
)
