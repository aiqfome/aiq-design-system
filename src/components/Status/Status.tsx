import React, { ReactNode } from 'react'
import PropTypes from 'prop-types'

import styled from 'styled-components'

import { Text } from '../Text'

export interface Props {
  children?: ReactNode
  statusColor?: string
}

const StatusStyled = styled(Text)<Props>`
  cursor: pointer;

  &::before {
    content: '';
    display: inline-block;
    width: 8px;
    height: 8px;
    margin-right: 10px;
    border-radius: 50%;
    background-color: ${({ statusColor, theme }) =>
      statusColor ? theme.colors[statusColor] : '#dedede'};
  }
`

export const Status: React.FC<Props> = ({ children, ...props }) => {
  return <StatusStyled {...props}>{children}</StatusStyled>
}

Status.propTypes = {
  children: PropTypes.node,
  statusColor: PropTypes.string
}
