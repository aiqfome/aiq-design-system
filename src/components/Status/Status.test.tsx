import React from 'react'
import renderer from 'react-test-renderer'

import { Status } from './Status'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Status />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
