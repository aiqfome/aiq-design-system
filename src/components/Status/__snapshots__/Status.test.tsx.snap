// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<span
  className="sc-bdnylx sc-gtssRu endPSk kGzeWi"
/>
`;
